### PR TITLE
Fix Range Get

### PR DESCRIPTION
**Pull request by windkit**
## Description
1. Remove duplicate `leo_gateway_rpc_handler:head`
2. Fix Ranges when Ending Offset is equal to or larger than object size
   ```
   If the last-byte-pos value is
   absent, or if the value is greater than or equal to the current
   length of the representation data, the byte range is interpreted as
   the remainder of the representation (i.e., the server replaces the
   value of last-byte-pos with a value that is one less than the current
   length of the selected representation).
   ```
   https://tools.ietf.org/html/rfc7233#section-2.1

### CODE REVIEW

**Comment by windkit** [comment]
Related to https://github.com/leo-project/leofs/issues/595
As Hadoop requests `{0, Object Length}` to LeoFS

File: apps/leo_gateway/src/leo_gateway_http_commons.erl [hunk 1]
```diff
@@ -1087,13 +1087,21 @@ get_range_object_small(_Req, BucketName, Key, Start, End,
...
-            get_body_length_1(Range, ObjectSize, 0);
-        {error, Reason} ->
-            {error, Reason}
-    end.
+fix_range_end(Range, ObjectSize) ->
+    fix_range_end_1(Range, ObjectSize, []).
+
+fix_range_end_1([], _, Acc) ->
+    lists:reverse(Acc);
+fix_range_end_1([{Start, End}|Rest], ObjectSize, Acc) when is_number(End),
```
Comment (mocchira) [inline apps/leo_gateway/src/leo_gateway_http_commons.erl]: s/is_number/is_integer/ as the range byte could not be floating point number.

File: apps/leo_gateway/src/leo_gateway_http_commons.erl [hunk 2]
```diff
@@ -1087,13 +1087,21 @@ get_range_object_small(_Req, BucketName, Key, Start, End,
...
+                                                           End >= ObjectSize ->
+    fix_range_end_1(Rest, ObjectSize, [{Start, ObjectSize - 1} | Acc]);
+fix_range_end_1([Head|Rest], ObjectSize, Acc) ->
+    fix_range_end_1(Rest, ObjectSize, [Head|Acc]).
+
```
Comment (mocchira) [inline apps/leo_gateway/src/leo_gateway_http_commons.erl]: Add a test case for the last byte `-ge` the size of the given object.

File: apps/leo_gateway/src/leo_gateway_http_commons.erl [hunk 3]
```diff
@@ -1087,13 +1087,21 @@ get_range_object_small(_Req, BucketName, Key, Start, End,
 
 
 %% @private
```
Comment (mocchira) [inline apps/leo_gateway/src/leo_gateway_http_commons.erl]: Add the comment about why this hack is needed (The description of this PR seems good to me

**Pull request merged**